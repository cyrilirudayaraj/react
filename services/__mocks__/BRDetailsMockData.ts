import { BRDetails } from '../../types';

export const BRDetailsMockData1: BRDetails = {
  ruleReportingCategoryId: '9',
  rules2TransformationStatus: '',
  internalFixText2:
    '<h2 style="text-align:left;"><span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 24px;font-family: DauphinPlain;">Where can I get some?</span></h2>\n<p style="text-align:justify;"><span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 14px;font-family: Open Sans", Arial, sans-serif;">There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don\'t look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn\'t anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.</span>&nbsp;</p>',
  ruleTypeId: '3',
  internalFixText1:
    '<p><strong>Lorem Ipsum</strong> <span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 14px;font-family: Open Sans;">is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</span>&nbsp;</p>',
  visitRuleDisplayLocation: 'VISITBILL_MEDICALCODES',
  contextName: 'CA - Prime Care - Inland Valley',
  contextId: '21',
  id: '1251',
  visitRuleDisplayLocationId: '2',
  businessRequirementTypeId: '2',
  legacyRuleId: '',
  rules2TransformationStatusId: '',
  name: 'Business Requirement Name',
  ruleReportingCategoryName: 'Coding: Procedure Code',
  description:
    '<p><span style="color: rgb(95,95,95);background-color: rgb(255,255,255);font-size: 14px;font-family: Source Sans Pro", arial, sans-serif;">Business Requirement Description (Client Friendly)</span>&nbsp;</p>',
  ruleType: 'CLAIMCREATE',
  localRuleUseCaseId: '2',
  localRuleUseCaseName: 'Capitation',
  businessRequirementType: 'Local',
  associatedBRTasks: [
    {
      id: '12',
      taskName: 'test1',
      statusId: '2',
      priorityId: '1',
      statusName: 'In Progress',
      escalatedYn: 'N',
      deploymentDate: '07/28/2021',
    },
    {
      id: '13',
      taskName: 'test2',
      statusId: '3',
      priorityId: '3',
      statusName: 'Assigned',
      escalatedYn: 'Y',
      deploymentDate: '07/29/2021',
    },
  ],
};
