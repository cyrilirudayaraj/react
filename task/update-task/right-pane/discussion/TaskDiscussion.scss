.create-comment {
  width: 100%;
  height: 100%;
  margin: auto;
  overflow: hidden;
  .comment-list {
    min-width: 100%;
    .form-group {
      min-width: 100%;
      .fe_c_form-field.fe_c_form-field--medium {
        padding-bottom: 0;
        margin-right: 0;
        padding-right: 0;
        min-width: 100%;
        .fe_l_grid-row.fe_l_grid-row--nested {
          min-width: 100%;
          .fe_l_grid-row__col.fe_l_grid-row__col--small-12.fe_l_grid-row__col--medium-12.fe_c_form-field__label.fe_c_form-field__label--forced-above {
            min-width: 100%;
          }
          .fe_l_grid-row__col.fe_l_grid-row__col--small-12.fe_l_grid-row__col--medium-11.fe_c_form-field__right.fe_c_form-field__right--label-forced-above {
            .fe_c_form-field__input-slot {
              min-width: 100%;
              .fe_c_textarea {
                min-width: 100%;
                display: flex;
              }
            }
          }
        }
      }
    }
  }
  .sendBtn {
    display: flex;
    position: fixed;
    bottom: 0;
    z-index: 99;
    right: 20px;
    justify-content: flex-end;
    #sendcomment {
      min-height: 28px;
      min-width: 114px;
      font-size: 12px;
      border-radius: 3px;
      letter-spacing: 0;
      text-align: center;
      margin-right: 2%;
    }
  }

  .user-comment-list-container {
    scrollbar-width: thin;
    height: 60vh;
    font-family: 'Source Sans Pro';
    font-size: 14px;
    letter-spacing: 0;
    line-height: 18px;
    overflow-y: auto;
    display: flex;
    flex-direction: column-reverse;
    margin-bottom: 20px;
    .user-comment-list-group {
      width: 100%;
      margin-bottom: 0;
      padding-bottom: 0;
      margin-left: 0;
      .user-container {
        width: 100%;
        list-style: none;
        padding-bottom: 0;
        margin-bottom: 0;
        .comment-date-container {
          width: 100%;
          margin-bottom: 20px;
          div {
            display: inline-block;
          }
          .borderright {
            padding-left: 5%;
          }
          .borderleft {
            padding-right: 5%;
          }
          .borderleft,
          .borderright {
            height: 1px;
            border-top: 0.5px solid #979797;
            margin-top: 8px;
          }
          .user-comment-list-group_date {
            text-align: center;
          }
        }
        .user-comment-result-list {
          padding-right: 15px;
          text-align: justify;
          .fe-center.highlight {
            background-color: #d9ebfa;
          }
          .fe-center {
            list-style: none;
            padding-bottom: 20px;
            margin-bottom: 0;
            .user-comment-result-date {
              color: #b5b5b5;
              line-height: 20px;
            }
            .user-comment-result-content {
              color: #333333;
              font-family: 'Source Sans Pro';
              font-size: 14px;
              letter-spacing: 0;
              line-height: 18px;
              padding-top: 14px;
              white-space: pre-wrap;
              white-space: -moz-pre-wrap;
              white-space: -pre-wrap;
              white-space: -o-pre-wrap;
              word-wrap: break-word;
            }
            span {
              padding-left: 0px;
              padding-right: 20px;
            }
          }
        }
      }
    }
    .user-comment-list-group {
      .user-container:last-child {
        .fe-center:last-child {
          padding-bottom: 0;
        }
      }
    }
  }
}
.user-comment-list-container::-webkit-scrollbar {
  width: 5px;
}
.user-comment-list-container::-webkit-scrollbar-thumb {
  border-radius: 10px;
}
