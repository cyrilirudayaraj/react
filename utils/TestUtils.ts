//TODO exclude this file during bundling
export function findByName(wrapper: any, name: string): any {
  const selector = "input[name='" + name + "']";
  return wrapper.find(selector).last();
}

export function findById(wrapper: any, id: string): any {
  const selector = '#' + id;
  return wrapper.find(selector).last();
}

export function changeValueById(wrapper: any, id: string, value: string): void {
  findById(wrapper, id).simulate('change', {
    target: { name: id, value: value },
  });
}

export function sleep(x: number): Promise<any> {
  return new Promise<void>((resolve) => {
    setTimeout(() => {
      resolve();
    }, x);
  });
}
