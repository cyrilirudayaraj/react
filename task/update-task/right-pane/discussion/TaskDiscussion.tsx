import React, { Component } from 'react';
import { connect } from 'react-redux';
import { TaskDiscusisonProps } from '../../../../../types';
import { getTaskDetail } from '../../../../../slices/TaskSlice';
import DiscussionForm from './DiscussionForm';
import { fetchUserCommentDetails } from '../../../../../slices/UserCommentSlice';
import './TaskDiscussion.scss';

class TaskDiscussion extends Component<TaskDiscusisonProps, any> {
  // TODO Use row component
  constructor(props: TaskDiscusisonProps) {
    super(props);
    this.props.fetchUserCommentDetails(this.props.taskId);
  }

  render(): JSX.Element {
    return (
      <div className="create-comment">
        <DiscussionForm taskdetails={this.props.task} {...this.props} />
      </div>
    );
  }
}

const mapTaskStateToProps = (state: any) => ({
  task: getTaskDetail(state),
});

export default connect(mapTaskStateToProps, {
  fetchUserCommentDetails,
})(TaskDiscussion);
