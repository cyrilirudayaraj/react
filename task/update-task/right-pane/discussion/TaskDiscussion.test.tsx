import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { TaskDetail } from '../../../../../types/index';
import TaskDiscussion from './TaskDiscussion';
import { Button, FormField } from '@athena/forge';
import { findById } from '../../../../../utils/TestUtils';
import { MockData } from '../../../../../services/__mocks__/MockData';
import moment from 'moment';
configure({ adapter: new Adapter() });

let wrapper: any;
let taskDetails: TaskDetail;
let store: any;
const userComments: any = [];
const TODAY = moment().clone().startOf('day').format('M/DD/YYYY hh:mm:ss');
const YESTERDAY = moment()
  .clone()
  .subtract(1, 'days')
  .startOf('day')
  .format('M/DD/YYYY hh:mm:ss');
const TODAYWITHOUTTIME = moment().clone().startOf('day').format('M/DD/YYYY');
const YESTERDAYWITHOUTTIME = moment()
  .clone()
  .subtract(1, 'days')
  .startOf('day')
  .format('M/DD/YYYY');
beforeEach(() => {
  taskDetails = {} as TaskDetail;
  taskDetails.activeTaskStepId = '16';
  taskDetails.id = '1';
  userComments[0] = {
    id: '1',
    taskId: '1',
    taskStepId: '16',
    content: 'Test cyril ontent 1',
    created: '08/15/2020',
    createdBy: 'jcyril',
    createdTimestamp: '08/15/2020 03:22:13',
  };
  userComments[1] = {
    id: '3',
    taskId: '1',
    taskStepId: '16',
    content: 'Test cyril content 2',
    created: YESTERDAYWITHOUTTIME,
    createdBy: 'jcyril',
    createdTimestamp: YESTERDAY,
  };
  userComments[2] = {
    id: '4',
    taskId: '1',
    taskStepId: '16',
    content: 'Test cyril content 3',
    created: TODAYWITHOUTTIME,
    createdBy: 'jcyril',
    createdTimestamp: TODAY,
  };
  store = configureStore([])({
    task: {
      taskDetails: taskDetails,
    },
    userPermission: {
      userPermissions: ['task.update'],
    },
    userComment: {
      userCommentDetails: userComments,
    },
    masterData: {
      users: MockData.USERS,
    },
    fetchUserCommentDetails: jest.fn(),
    setCommentDetails: jest.fn(),
  });
  store.dispatch = jest.fn();
});
jest.mock('draft-js/lib/generateRandomKey', () => () => '123');

describe('Task Discussion', () => {
  it('basic snapshot', () => {
    wrapper = mount(
      <Provider store={store}>
        <TaskDiscussion commentId="" taskId="" resetCommentId="" />
      </Provider>
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('textarea renders properly', () => {
    expect(wrapper.find(FormField)).toHaveLength(1);
  });

  it('Button renders properly', () => {
    expect(wrapper.find(Button)).toHaveLength(1);
  });

  it('comment submission ', () => {
    findById(wrapper, 'content').simulate('change', {
      target: { value: 'Testing comment section' },
    });
    findById(wrapper, 'sendcomment').simulate('click');
  });
});
