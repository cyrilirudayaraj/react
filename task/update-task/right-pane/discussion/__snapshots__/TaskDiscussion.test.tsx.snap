// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Discussion basic snapshot 1`] = `
"<Provider store={{...}}>
  <Connect(TaskDiscussion) commentId=\\"\\" taskId=\\"\\" resetCommentId=\\"\\">
    <TaskDiscussion commentId=\\"\\" taskId=\\"\\" resetCommentId=\\"\\" task={{...}} fetchUserCommentDetails={[Function (anonymous)]}>
      <div className=\\"create-comment\\">
        <Connect(DiscussionForm) taskdetails={{...}} commentId=\\"\\" taskId=\\"\\" resetCommentId=\\"\\" task={{...}} fetchUserCommentDetails={[Function (anonymous)]}>
          <DiscussionForm taskdetails={{...}} commentId=\\"\\" taskId=\\"\\" resetCommentId=\\"\\" task={{...}} fetchUserCommentDetails={[Function (anonymous)]} comments={{...}} createUserComment={[Function (anonymous)]}>
            <Formik enableReinitialize={true} initialValues={{...}} onSubmit={[Function: onSubmit]} validationSchema={[undefined]} validateOnChange={false} validateOnBlur={false} validateOnMount={false}>
              <UserCommentList usercomment={{...}} taskdetails={{...}} commentId=\\"\\" taskId=\\"\\" resetCommentId=\\"\\" task={{...}} fetchUserCommentDetails={[Function (anonymous)]} comments={{...}} createUserComment={[Function (anonymous)]}>
                <div id=\\"user-comment-list-container\\" className=\\"user-comment-list-container\\">
                  <ul className=\\"user-comment-list-group\\">
                    <li className=\\"user-container\\">
                      <GridRow className=\\"comment-date-container\\" removeGuttersSmall={true}>
                        <div className=\\"fe_l_grid-row fe_l_grid-row--no-gutters-small comment-date-container\\">
                          <GridCol className=\\"borderleft\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-5 borderleft\\" />
                          </GridCol>
                          <GridCol className=\\"user-comment-list-group_date\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-2 user-comment-list-group_date\\">
                              8/15/2020
                            </div>
                          </GridCol>
                          <GridCol className=\\"borderright\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-5 borderright\\" />
                          </GridCol>
                        </div>
                      </GridRow>
                      <div className=\\"user-comment-result-list\\">
                        <div id=\\"1\\" className=\\"fe-center\\">
                          <span className=\\"user-comment-result-name\\">
                            jcyril
                          </span>
                          <span className=\\"user-comment-result-date\\">
                            8/15/2020 3:22 AM
                          </span>
                          <div className=\\"user-comment-result-content\\" dangerouslySetInnerHTML={{...}} />
                        </div>
                      </div>
                    </li>
                    <li className=\\"user-container\\">
                      <GridRow className=\\"comment-date-container\\" removeGuttersSmall={true}>
                        <div className=\\"fe_l_grid-row fe_l_grid-row--no-gutters-small comment-date-container\\">
                          <GridCol className=\\"borderleft\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-5 borderleft\\" />
                          </GridCol>
                          <GridCol className=\\"user-comment-list-group_date\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-2 user-comment-list-group_date\\">
                              Yesterday
                            </div>
                          </GridCol>
                          <GridCol className=\\"borderright\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-5 borderright\\" />
                          </GridCol>
                        </div>
                      </GridRow>
                      <div className=\\"user-comment-result-list\\">
                        <div id=\\"3\\" className=\\"fe-center\\">
                          <span className=\\"user-comment-result-name\\">
                            jcyril
                          </span>
                          <span className=\\"user-comment-result-date\\">
                            Yesterday 12:00 PM
                          </span>
                          <div className=\\"user-comment-result-content\\" dangerouslySetInnerHTML={{...}} />
                        </div>
                      </div>
                    </li>
                    <li className=\\"user-container\\">
                      <GridRow className=\\"comment-date-container\\" removeGuttersSmall={true}>
                        <div className=\\"fe_l_grid-row fe_l_grid-row--no-gutters-small comment-date-container\\">
                          <GridCol className=\\"borderleft\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-5 borderleft\\" />
                          </GridCol>
                          <GridCol className=\\"user-comment-list-group_date\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-2 user-comment-list-group_date\\">
                              Today
                            </div>
                          </GridCol>
                          <GridCol className=\\"borderright\\" width={{...}}>
                            <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-5 borderright\\" />
                          </GridCol>
                        </div>
                      </GridRow>
                      <div className=\\"user-comment-result-list\\">
                        <div id=\\"4\\" className=\\"fe-center\\">
                          <span className=\\"user-comment-result-name\\">
                            jcyril
                          </span>
                          <span className=\\"user-comment-result-date\\">
                            12:00 PM
                          </span>
                          <div className=\\"user-comment-result-content\\" dangerouslySetInnerHTML={{...}} />
                        </div>
                      </div>
                    </li>
                  </ul>
                </div>
              </UserCommentList>
              <UserCommentForm formik={{...}}>
                <Form labelAlwaysAbove={true} includeSubmitButton={false} className=\\"comment-list\\" buttonText=\\"Submit\\" noValidate={true} requiredVariation=\\"blueBarWithRequiredLabel\\" layout=\\"medium\\" nested={true} labelWidth={3}>
                  <form onSubmit={[undefined]} className=\\"comment-list\\" noValidate={true}>
                    <div className=\\"form-group\\">
                      <Connect(RichTextEditor) id=\\"content\\" wrapperId={5} formik={{...}} className=\\"discussion_text_box\\" enableShowMore={false} enableTagging={true}>
                        <RichTextEditor id=\\"content\\" wrapperId={5} formik={{...}} className=\\"discussion_text_box\\" enableShowMore={false} enableTagging={true} usersInfo={{...}} fetchUsersOnce={[Function (anonymous)]} labelText=\\"\\" maxLength={4000} isRequired={false}>
                          <div className=\\"textarea-component\\">
                            <FormField id=\\"content\\" wrapperId={5} labelText=\\"\\" labelWidth={1} className=\\"rte\\" inputAs={[Function: r]} required={true} wrapperClassName=\\"discussion_text_box\\" editorClassName=\\"editor-textarea\\" editorState={{...}} onEditorStateChange={[Function (anonymous)]} readOnly={[undefined]} wrapperStyle={{...}} toolbarHidden={true} error={[undefined]} toolbar={{...}} onChange={[Function (anonymous)]} onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]} mention={{...}}>
                              <FormFieldLayout labelWidth={1} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"rte\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
                                <div className=\\"fe_c_form-field fe_c_form-field--medium rte\\">
                                  <GridRow nested={true}>
                                    <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                      <GridCol width={{...}} className=\\"fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                        <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-12 fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                          <Label id=\\"content-label\\" htmlFor=\\"content\\" text=\\"\\" disabled={[undefined]} required={true} useLegend={false}>
                                            <label htmlFor=\\"content\\" id=\\"content-label\\" className=\\"fe_c_label\\" />
                                          </Label>
                                        </div>
                                      </GridCol>
                                      <GridCol width={{...}} className=\\"fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                        <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-11 fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                          <div className=\\"fe_c_form-field__input-slot\\">
                                            <r wrapperId={5} wrapperClassName=\\"discussion_text_box\\" editorClassName=\\"editor-textarea\\" editorState={{...}} onEditorStateChange={[Function (anonymous)]} readOnly={[undefined]} wrapperStyle={{...}} toolbarHidden={true} toolbar={{...}} onChange={[Function (anonymous)]} onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]} mention={{...}} className=\\"fe_c_form-field__input\\" id=\\"content\\" name=\\"content\\" required={true} hideRequiredStyles={false} disabled={[undefined]} error={[undefined]} aria-labelledby=\\"content-label\\" toolbarOnFocus={false} stripPastedStyles={false} localization={{...}} customDecorators={{...}}>
                                              <div id=\\"rdw-wrapper-5\\" className=\\"discussion_text_box rdw-editor-wrapper\\" style={{...}} onClick={[Function (anonymous)]} onBlur={[Function (anonymous)]} aria-label=\\"rdw-wrapper\\">
                                                <div className=\\"editor-textarea rdw-editor-main\\" style={[undefined]} onClick={[Function (anonymous)]} onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]} onKeyDown={[Function: onKeyDown]} onMouseDown={[Function (anonymous)]}>
                                                  <DraftEditor keyBindingFn={[Function (anonymous)]} editorState={{...}} onChange={[Function (anonymous)]} blockStyleFn={[Function: D]} customStyleMap={{...}} handleReturn={[Function (anonymous)]} handlePastedText={[Function (anonymous)]} blockRendererFn={[Function (anonymous)]} handleKeyCommand={[Function (anonymous)]} ariaLabel=\\"rdw-editor\\" blockRenderMap={{...}} wrapperId={5} readOnly={false} className=\\"fe_c_form-field__input\\" id=\\"content\\" name=\\"content\\" required={true} hideRequiredStyles={false} disabled={[undefined]} error={[undefined]} aria-labelledby=\\"content-label\\" stripPastedStyles={false} ariaDescribedBy=\\"{{editor_id_placeholder}}\\" spellCheck={false}>
                                                    <div className=\\"DraftEditor-root\\">
                                                      <div className=\\"DraftEditor-editorContainer\\">
                                                        <div aria-activedescendant={[undefined]} aria-autocomplete={[undefined]} aria-controls={[undefined]} aria-describedby={[undefined]} aria-expanded={{...}} aria-label=\\"rdw-editor\\" aria-labelledby={[undefined]} aria-multiline={[undefined]} aria-owns={[undefined]} autoCapitalize={[undefined]} autoComplete={[undefined]} autoCorrect={[undefined]} className=\\"notranslate public-DraftEditor-content\\" contentEditable={true} data-testid={[undefined]} onBeforeInput={[Function (anonymous)]} onBlur={[Function (anonymous)]} onCompositionEnd={[Function (anonymous)]} onCompositionStart={[Function (anonymous)]} onCopy={[Function (anonymous)]} onCut={[Function (anonymous)]} onDragEnd={[Function (anonymous)]} onDragEnter={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onDragOver={[Function (anonymous)]} onDragStart={[Function (anonymous)]} onDrop={[Function (anonymous)]} onFocus={[Function (anonymous)]} onInput={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyPress={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onPaste={[Function (anonymous)]} onSelect={[Function (anonymous)]} role=\\"textbox\\" spellCheck={false} style={{...}} suppressContentEditableWarning={true} tabIndex={[undefined]}>
                                                          <UpdateDraftEditorFlags editor={{...}} editorState={{...}} />
                                                          <DraftEditorContents blockRenderMap={{...}} blockRendererFn={[Function (anonymous)]} blockStyleFn={[Function: D]} customStyleMap={{...}} customStyleFn={[undefined]} editorKey=\\"123\\" editorState={{...}} preventScroll={[undefined]} textDirectionality={[undefined]}>
                                                            <div data-contents=\\"true\\">
                                                              <div className=\\"\\" data-block={true} data-editor=\\"123\\" data-offset-key=\\"123-0-0\\">
                                                                <DraftEditorBlock contentState={{...}} block={{...}} blockProps={[undefined]} blockStyleFn={[Function: D]} customStyleMap={{...}} customStyleFn={[undefined]} decorator={{...}} direction=\\"LTR\\" forceSelection={false} offsetKey=\\"123-0-0\\" preventScroll={[undefined]} selection={{...}} tree={{...}}>
                                                                  <div data-offset-key=\\"123-0-0\\" className=\\"public-DraftStyleDefault-block public-DraftStyleDefault-ltr\\">
                                                                    <DraftEditorLeaf offsetKey=\\"123-0-0\\" block={{...}} start={0} selection={{...}} forceSelection={false} text=\\"\\" styleSet={{...}} customStyleMap={{...}} customStyleFn={[undefined]} isLast={true}>
                                                                      <span data-offset-key=\\"123-0-0\\" style={{...}}>
                                                                        <DraftEditorTextNode>
                                                                          <br data-text=\\"true\\" />
                                                                        </DraftEditorTextNode>
                                                                      </span>
                                                                    </DraftEditorLeaf>
                                                                  </div>
                                                                </DraftEditorBlock>
                                                              </div>
                                                            </div>
                                                          </DraftEditorContents>
                                                        </div>
                                                      </div>
                                                    </div>
                                                  </DraftEditor>
                                                </div>
                                              </div>
                                            </r>
                                          </div>
                                          <div className=\\"fe_c_form-field__status\\" />
                                        </div>
                                      </GridCol>
                                    </div>
                                  </GridRow>
                                </div>
                              </FormFieldLayout>
                            </FormField>
                          </div>
                        </RichTextEditor>
                      </Connect(RichTextEditor)>
                    </div>
                    <div className=\\"sendBtn\\">
                      <Connect(WFEnableForPermission) permission=\\"discussion.create\\">
                        <WFEnableForPermission permission=\\"discussion.create\\" userPermissions={{...}} dispatch={[Function: mockConstructor]}>
                          <ForwardRef id=\\"sendcomment\\" className=\\"primary-button\\" text=\\"Send\\" onClick={[Function (anonymous)]} variant=\\"primary\\" disabled={false} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                            <button id=\\"sendcomment\\" className=\\"fe_c_button fe_c_button--primary fe_c_button--medium primary-button\\" aria-disabled={false} aria-label={[undefined]} onClick={[Function: handleClick]} type=\\"button\\">
                              <div className=\\"fe_c_button__content\\">
                                <span className=\\"fe_c_button__text fe_c_button__text--medium fe_c_button__text--primary\\">
                                  Send
                                </span>
                              </div>
                            </button>
                          </ForwardRef>
                        </WFEnableForPermission>
                      </Connect(WFEnableForPermission)>
                    </div>
                  </form>
                </Form>
              </UserCommentForm>
            </Formik>
          </DiscussionForm>
        </Connect(DiscussionForm)>
      </div>
    </TaskDiscussion>
  </Connect(TaskDiscussion)>
</Provider>"
`;
